skeleton/ucp/main_module.php.twig

```diff
@@ -21,34 +21,34 @@ class main_module
 	{
 		global $db, $request, $template, $user;
 
-		$this->tpl_name = 'ucp_demo_body';
-		$this->page_title = $user->lang('UCP_DEMO_TITLE');
-		add_form_key('acme/demo');
+		$this->tpl_name = 'ucp_{{ EXTENSION.extension_name|lower }}_body';
+		$this->page_title = $user->lang('UCP_{{ EXTENSION.extension_name|upper }}_TITLE');
+		add_form_key('{{ EXTENSION.vendor_name|lower }}_{{ EXTENSION.extension_name|lower }}');
 
 		$data = array(
-			'user_acme' => $request->variable('user_acme', $user->data['user_acme']),
+			'user_{{ EXTENSION.extension_name|lower }}' => $request->variable('user_{{ EXTENSION.extension_name|lower }}', $user->data['user_{{ EXTENSION.extension_name|lower }}']),
 		);
 
 		if ($request->is_set_post('submit'))
 		{
-			if (!check_form_key('acme/demo'))
+			if (!check_form_key('{{ EXTENSION.vendor_name|lower }}_{{ EXTENSION.extension_name|lower }}'))
 			{
 				trigger_error($user->lang('FORM_INVALID'));
 			}
 
 			$sql = 'UPDATE ' . USERS_TABLE . '
 				SET ' . $db->sql_build_array('UPDATE', $data) . '
-				WHERE user_id = ' . $user->data['user_id'];
+				WHERE user_id = ' . (int) $user->data['user_id'];
 			$db->sql_query($sql);
 
 			meta_refresh(3, $this->u_action);
-			$message = $user->lang('UCP_DEMO_SAVED') . '<br /><br />' . $user->lang('RETURN_UCP', '<a href="' . $this->u_action . '">', '</a>');
+			$message = $user->lang('UCP_{{ EXTENSION.extension_name|upper }}_SAVED') . '<br /><br />' . $user->lang('RETURN_UCP', '<a href="' . $this->u_action . '">', '</a>');
 			trigger_error($message);
 		}
 
 		$template->assign_vars(array(
-			'S_USER_ACME'	=> $data['user_acme'],
-			'S_UCP_ACTION'	=> $this->u_action,
+			'S_USER_{{ EXTENSION.extension_name|upper }}'	=> $data['user_{{ EXTENSION.extension_name|lower }}'],
+			'U_UCP_ACTION'		=> $this->u_action,
```

Comment:
One tab too many added here.

Reply:
Not too sure what you mean on this one?

Reply:
You added an extra tab after the `U_UCP_ACTION ` lang key

Reply:
This is irrelevant I suppose since, because the line above can be of varying size, it's pointless to add tabs to align anything here, but for simplicity, the `U_UCP_ACTION` should just have 1 tab before the `=>` and end users can change it if  they like on their own, based on the size of their `S_USER_` lang key.